### ISSUE TITLE
Provide API Authentication/Registration

### ISSUE THREAD

**Issue by willisjtc** [keycloak/keycloak#11992]
### Description

Would be great to have a custom login page that doesn't redirect to Keycloak's login page.

Because of this, keycloak would need to provide httpOnly, secure tokens to the frontend.

### Motivation

All major auth providers provide away to use an api to login, register, or get access tokens. Time for keycloak to provide that mechanism as well

**Comment by JDDurrant** [comment]
Agreed. I really like Keycloak, but the absence of this feature is a real deal breaker for me.
